File: backend/app/schemas/location.py
from pydantic import BaseModel
from typing import Optional


class Location(BaseModel):
    location_text: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    location_source: Optional[str] = None


class LocationResponse(Location):
    pass


class LocationSuggestion(BaseModel):
    id: str
    name: str
    city: str
    country: str
    latitude: Optional[float] = None
    longitude: Optional[float] = None

File: frontend/src/types/location.ts
export interface Location {
  location_text?: string;
  latitude?: number;
  longitude?: number;
  location_source?: string;
}

export interface LocationSuggestion {
  id: string;
  name: string;
  city: string;
  country: string;
  latitude?: number;
  longitude?: number;
}